Src/PerformanceCollector/Shared/Implementation/WebAppPerformanceCollector/NormalizedCPUPercentageGauge.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.WebAppPerformanceCollector
+{
+    using System;
+    using System.Globalization;
+
+    /// <summary>
+    /// Gauge that computes normalized CPU percentage utilized by a process by utilizing the last computed time (divided by the processors count).
+    /// </summary>
+    internal class NormalizedCPUPercentageGauge : CPUPercenageGauge
+    {
+        /// <summary>Specific environment variable for Azure App Services.</summary>
+        private const string ProcessorsCounterEnvironmentVariable = "NUMBER_OF_PROCESSORS";
+
+        private readonly object lockObject = new object();
+        private bool isInitialized = false;
+        private int processorsCount = -1;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NormalizedCPUPercentageGauge"/> class.
+        /// </summary>
+        /// <param name="name"> Name of the SumUpCountersGauge.</param>
+        /// <param name="value"> Gauges to sum.</param>
+        public NormalizedCPUPercentageGauge(string name, ICounterValue value) : base(name, value)
+        {
+        }
+
+        /// <summary>
+        /// Returns the normalized percentage of the CPU process utilization time divided by the number of processors with respect to the total duration.
+        /// </summary>
+        /// <returns>The value of the target metric.</returns>
+        protected override double Collect()
+        {
+            if (!this.isInitialized)
+            {
+                lock (this.lockObject)
+                {
+                    if (!this.isInitialized)
+                    {
+                        this.processorsCount = this.GetProcessorsCount();
```

Comment:
Could you please not throw - change standard counters instead of aligning with them. Exceptions from performance counters collection are dominant in the exceptions statistics feature.